pallene/types.lua

```diff
@@ -194,7 +198,8 @@ local function equivalent(t1, t2, is_gradual)
         return t1 == t2
 
     else
-        return error("impossible")
+        return typedecl.tag_error(tag1,
+            string.format("attempt to check equivalence of types %s and %s.", tag1, tag2))
```

Comment:
Oh, I noticed that in places where `string.matches` is used, the prefix being tested is `"types.T."`.
I think it would make more sense to check every tag prefix with the the second "." included, no ?